libs/prelude/Prelude/Interfaces.idr

```diff
@@ -361,7 +361,7 @@ for_ = flip traverse_
 ||| Note: In Haskell, `choice` is called `asum`.
 public export
 choice : (Foldable t, Alternative f) => t (f a) -> f a
-choice = foldr (<|>) empty
+choice = foldr (\ x, xs => x <|> xs) empty
```

Comment:
Despite `<|>` is lazy on the right argument, this implementation of `choice` is fully eager. This thing needs to be revised as soon as the problem with the trick of lazy folding is solved.